wallet/src/repository/wallet/mod.rs

```diff
@@ -1266,17 +1293,20 @@ where
             max_weight,
         )?;
 
-        let change_pkh =
-            self.calculate_change_address(dr_output.is_none(), &tx_info.inputs, state)?;
+        let change_pkh = self.calculate_change_address(state, None)?;
```

Comment:
This is always `None`, I believe it should be `Some` when the transaction is a data request to match the old behavior.

Reply:
Yes, trying to deploy data requests from sheikah is broken, with this error:

```
[2022-10-25T14:46:31Z ERROR witnet_wallet::actors::app::error] Node Error: request failed: RPC error: Error { code: InternalError, message: "Public key hash mismatch: expected twit1v39juf5kvqapku258rdwlpch3p5zh8dt93nn0h got twit185j3rmer0kfhzm3uy3ecv2d495wge7n228cfcp", data: None }
```

Reply:
Ooops I knew I was breaking it, but forgot to put a TODO in place so I wouldn't forget to fix it later :sweat_smile: 

It should be fixed now!